java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/sink/RedisSinkFunction.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+
+        this.indexBuffer = ByteBuffer.allocate(4).order(ByteOrder.BIG_ENDIAN);
+        this.timestampBuffer = ByteBuffer.allocate(8).order(ByteOrder.BIG_ENDIAN);
+
+        jedis = new Jedis(host, port);
+        if (!StringUtils.isNullOrWhitespaceOnly(password)) {
+            jedis.auth(password);
+        }
+        jedis.select(dbNum);
+        if (timestampFieldIndex >= 0) {
+            evalScriptSHA = jedis.scriptLoad(this.evalScript);
+        }
+    }
+
+    @Override
+    public void invoke(RowData data, Context context) {
+        byte[] originalKey = data.getBinary(keyFieldIndex);
+        byte[] key = new byte[keyPrefix.length + originalKey.length];
+        System.arraycopy(keyPrefix, 0, key, 0, keyPrefix.length);
+        System.arraycopy(originalKey, 0, key, keyPrefix.length, originalKey.length);
+
+        Map<byte[], byte[]> value = new HashMap<>();
+        int arity = data.getArity();
+        for (int i = 0; i < arity; i++) {
+            if (timestampFieldIndex == i) {
+                // timestamp values have specific serialization strategy
+                continue;
+            }
+
+            if (keyFieldIndex == i) {
+                // don't store key values
+                continue;
+            }
+
+            byte[] fieldValue = data.getBinary(i);
+
+            if (fieldValue == null) {
+                // don't store null values
+                continue;
+            }
+
+            indexBuffer.putInt(0, i);
+            value.put(indexBuffer.array(), fieldValue);
+        }
+
+        if (timestampFieldIndex < 0) {
+            if (value.isEmpty()) {
+                return;
+            }
+            this.jedis.hset(key, value);
+        } else {
+            timestampBuffer.putLong(0, data.getLong(timestampFieldIndex));
+
+            List<byte[]> arguments = new ArrayList<>();
+            arguments.add(key);
+            arguments.add(timestampBuffer.array());
+            for (Map.Entry<byte[], byte[]> entry : value.entrySet()) {
+                arguments.add(entry.getKey());
+                arguments.add(entry.getValue());
+            }
+
+            this.jedis.evalsha(this.evalScriptSHA, Collections.emptyList(), arguments);
+        }
+    }
+
+    @Override
+    public void close() {
+        // TODO: Remove the registered script from Redis when Redis supports removing a certain
+        // script.
+        jedis.close();
+    }
+
+    private static byte[] checkAndGetKeyPrefix(String namespace) {
+        Preconditions.checkArgument(
+                namespace.matches("^[A-Za-z0-9][A-Za-z0-9_]*$"),
+                "Namespace %s should only contain letters, numbers and underscore, "
+                        + "and should not start with underscore.");
+
+        return (namespace + ":").getBytes(StandardCharsets.UTF_8);
```

Comment:
Would it be a bit more readable to use `.` as the separator? 

The key would look like `namespace_a.original_key`.